runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -2960,8 +2960,18 @@ TR::Register *J9::X86::TreeEvaluator::ArrayStoreCHKEvaluator(TR::Node *node, TR:
    doneLabel = generateLabelSymbol(cg);
    doneLabel->setEndInternalControlFlow();
```

Comment:
Rather than gating this deferred evaluation on repeated checks for off heap, I think the code might be more readable if you used a different variable altogether and used that throughout.  It would also allow you to turn this deferral off entirely, should this be a useful diagnostic feature.  e.g.,
```
    static char *disableDeferDestinationEvaluation = feGetEnv("TR_DisableDeferDestinationEvaluation");
    bool deferDestinationEvaluation = TR::Compiler->om.isOffHeapAllocationEnabled() && !disableDeferDestinationEvaluation
```

Reply:
Added disabling defer option.